commons/src/main/java/io/ballerina/stdlib/graphql/commons/utils/SdlSchemaStringGenerator.java

```diff
@@ -216,10 +223,23 @@ private static String createArgs(List<InputValue> inputValues) {
         if (inputValues.isEmpty()) {
             return EMPTY_STRING;
         }
-        for (InputValue arg : inputValues) {
-            args.add(getFormattedString(ARGS_VALUE_FORMAT, arg.getName(), createArgType(arg)));
+        boolean hasDescription = inputValues.stream()
+                .anyMatch(input -> input.getDescription() != null && !input.getDescription().isEmpty());
+        if (hasDescription) {
+            for (InputValue arg : inputValues) {
+                args.add(getFormattedString(ARGS_VALUE_FORMAT, createArgDescription(arg.getDescription()),
+                        DOUBLE_INDENTATION, arg.getName(), createArgType(arg)));
+            }
+            return getFormattedString(ARGS_FORMAT, LINE_SEPARATOR, String.join(LINE_SEPARATOR, args),
+                    LINE_SEPARATOR, INDENTATION);
+        } else {
+            for (InputValue arg : inputValues) {
+                args.add(getFormattedString(ARGS_VALUE_FORMAT, EMPTY_STRING, EMPTY_STRING, arg.getName(),
+                        createArgType(arg)));
+            }
+            return getFormattedString(ARGS_FORMAT, EMPTY_STRING, String.join(COMMA_SIGN + SPACE, args), EMPTY_STRING,
+                    EMPTY_STRING);
         }
```

Comment:
Shouldn't this be handled in the `createArgDescription` method? Here, the `inputValues` array is iterated twice, once to check `hasDescription`, another time to get the description. We can avoid this by handling it inside the description generation.

Reply:
If there is a single argument description, the entire argument block format and each single argument format will be changed. Therefore used two iterations here. If this is moved into `createDescription`, it will be complex to handle this scenario.

Reply:
Can you include an example of that?

Reply:
1. if there is no arg descriptions
`name(id: String!, age: Int!, isAdult: Boolean!): String!`

2. if there is at least one arg description
```
name(
    id: String!
    """
    This should be an integer
    This is a Non-Null type argument
    """
    age: Int!
    isAdult: Boolean!
  ): String!
```